platform/mellanox/mlnx-platform-api/sonic_platform/component.py

```diff
@@ -0,0 +1,145 @@
+#!/usr/bin/env python
+
+#############################################################################
+# Mellanox
+#
+# implementation of new platform api
+#############################################################################
+
+try:
+    from sonic_platform_base.component_base import ComponentBase
+    from glob import glob
+    import subprocess
+    import io
+    import re
+except ImportError as e:
+    raise ImportError(str(e) + "- required module not found")
+
+#components definitions
+COMPONENT_BIOS = "BIOS"
+COMPONENT_CPLD = "CPLD"
+
+BIOS_QUERY_VERSION_COMMAND = 'dmidecode -t 11'
+CPLD_VERSION_FILE_PATTERN = '/var/run/hw-management/system/cpld[0-9]_version'
+CPLD_VERSION_MAX_LENGTH = 4
+
+class Component(ComponentBase):
+    def get_name(self):
+        """
+        Retrieves the name of the component
+
+        Returns:
+            A string containing the name of the component
+        """
+        return self.name
+
+
+    def _read_generic_file(self, filename, len):
+        """
+        Read a generic file, returns the contents of the file
+        """
+        result = ''
+        try:
+            with io.open(filename, 'r') as fileobj:
+                result = fileobj.read(len)
+            return result
+        except Exception as e:
```

Comment:
@stephenxs IMHO, this is critical so the exception should be handled in a proper way by upper layer.